### PR TITLE
Nefi scripts

### PR DESCRIPTION
**Pull request by Luke-Dramko**
## Description

This collection of scripts help run PEcAn every morning and plug directly into a CRON job.  They are not part of the workflow, but I've found them useful or necessary for developing and running iterative forecasting.  All of these files are in `pecan/modules/assim.sequential/inst/`.  The request was to put these here so they wouldn't be lost.

For reference, the CRON job looks like:
`0 3 * * * /absolute/path/to/script/run.gefs.sipnet.sh`

## Motivation and Context

These scripts are part of the NEFI initiative and aim to iteratively call the PEcAn workflow, including its SDA section.

## Review Time Estimate

- [ ] Immediately
- [x] Within one week
- [ ] When possible
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/assim.sequential/inst/NEFI/generate.gefs.xml.R [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+# but good practice anyway).
+settings$run$site$met.start <- format(Sys.time() - lubridate::days(12), "%Y-%m-%d")
+settings$run$site$met.end <- format(Sys.time() + lubridate::days(16), "%Y-%m-%d")
+
+settings$ensemble$start.year <- format(start_date, "%Y")
+settings$ensemble$end.year <- as.character(end_date, "%Y")
+
+# Determine the workflow ID by looking in the database
+ids <- PEcAn.DB::db.query("SELECT id FROM workflows", con)
+ids <- ids[ids<2000000000] # There are several placeholder ids that are much too large in the database, so this screens
```
Comment (mdietze) [inline modules/assim.sequential/inst/NEFI/generate.gefs.xml.R]: FYI, it's more efficient to have SQL do this sort of filtering than to query a large table and then post-process it in R

File: modules/assim.sequential/inst/NEFI/generate.gefs.xml.R [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                           # those out.  Kind of hack-y, but it works.
+workflowid <- as.character(max(ids) + 1)
+settings$workflow$id <- workflowid
```
Comment (mdietze) [inline modules/assim.sequential/inst/NEFI/generate.gefs.xml.R]: Don't do this. Create the record in the workflow table, then have that insert return the new ID. This is much much safer than what you're doing here, which is creating the ID of the record before you create the record.

File: modules/assim.sequential/inst/NEFI/generate.gefs.xml.R [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+# Create the output directory.  PEcAn does not do this for you.  It's normally the job of the
+# web interface's php code.
+if (!dir.exists(outdir)) {
+  dir.create(outdir, recursive=TRUE, showWarnings = FALSE)
+}
+
+# Register new workflow id with the database
+hostname <- PEcAn.remote::fqdn()
+query <- paste0("INSERT INTO workflows (site_id, model_id, notes, folder, hostname, start_date,",
```
Comment (mdietze) [inline modules/assim.sequential/inst/NEFI/generate.gefs.xml.R]: This should have a `RETURNING id` at the end and doesn't need the started_at and created_at

Comment (Luke-Dramko) [reply modules/assim.sequential/inst/NEFI/generate.gefs.xml.R]: Looks like all three requested changes have to do with the workflow id/database.  I didn't know the query returned with workflow id after insertion; that's certainly much better.  Not a hard change to make!

File: modules/assim.sequential/inst/NEFI/generate.gefs.xml.R (hunk 1 above)
Comment (infotroph) [reply modules/assim.sequential/inst/NEFI/generate.gefs.xml.R]: Also, IDs over 2e9 aren't placeholders, they're from [other nodes in the BETY network](https://pecan.gitbook.io/betydb-documentation/betydb-system-administration/distributed-betydb#primary-key-allocations). As written, this won't work for any workflow created outside the Dietze or Serbin labs.

File: modules/assim.sequential/inst/NEFI/generate.gefs.xml.R (hunk 3 above)
Comment (Luke-Dramko) [reply modules/assim.sequential/inst/NEFI/generate.gefs.xml.R]: Workflow ID bad practices have been fixed.

**Pull request merged**